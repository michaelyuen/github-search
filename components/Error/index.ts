export * from "./Error";
export * from "./styles";
