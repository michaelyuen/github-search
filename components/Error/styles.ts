import styled from "styled-components";

export const ErrorContainer = styled.p`
  color: var(--error-color);
`;
