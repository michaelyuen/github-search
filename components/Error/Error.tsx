import { ErrorContainer } from "./styles";

export interface ErrorProps {
  message: string;
}

export const Error: React.FC<ErrorProps> = ({ message, ...rest }) => {
  return <ErrorContainer {...rest}>{message}</ErrorContainer>;
};
