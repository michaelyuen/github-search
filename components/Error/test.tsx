import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import { Error } from "./Error";

describe("Error", () => {
  test("renders", () => {
    render(<Error message="Something went wrong" />);
    expect(screen.getByText("Something went wrong")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(<Error message="Something went wrong" />);
    expect(container).toMatchSnapshot();
  });

  test("has no a11y violations", async () => {
    const { container } = render(<Error message="Something went wrong" />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
