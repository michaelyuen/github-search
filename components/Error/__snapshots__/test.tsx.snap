// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error matches snapshot 1`] = `
<div>
  <p
    class="sc-bcXHqe bbsUvP"
  >
    Something went wrong
  </p>
</div>
`;
